### PR TITLE
Try fixing the 'brew update' issue in bootstrap script

### PR DESCRIPTION
**Pull request by daxian-dbw**
This is an attempt to fix the 'brew update' failure that happens a lot in your macOS CI runs: `RPC failed; curl 56 SSLRead() return error -36`.

With this change, we will try run `brew update` twice. The first run is the same as before, but if it fails, then we increase the Git buffer size before the second attempt.

### CODE REVIEW

File: tools/installpsh-osx.sh [hunk 1]
```diff
@@ -127,10 +127,22 @@ fi
...
+    if [[ $count == 2 ]]; then
+        echo "ERROR: Refreshing Homebrew cache failed..." >&2
+        exit 2
+    fi
+
+    # The update failed for the first try. An error we see a lot in our CI is "RPC failed; curl 56 SSLRead() return error -36".
+    # A potential solution is to increase the Git buffer size to a larger number, say 150 mb. The default buffer size is 1 mb.
+    echo "First attempt of update failed. Increase Git buffer size and try again ..."
+    git config --global http.postBuffer 157286400
+    sleep 5
```
Comment (adityapatwardhan) [inline tools/installpsh-osx.sh]: Why is this required? Can you put a comment.

**Review by adityapatwardhan — APPROVED** [review]
With minor comment.

File: tools/installpsh-osx.sh (hunk 1 above)
Comment (daxian-dbw) [reply tools/installpsh-osx.sh]: Sure, will add a comment.

File: tools/installpsh-osx.sh [hunk 2]
```diff
@@ -127,10 +127,22 @@ fi
...
 echo "Refreshing Homebrew cache..."
-if ! brew update > /dev/null; then
-    echo "ERROR: Refreshing Homebrew cache failed..." >&2
-    exit 2
-fi
+for count in {1..2}; do
+    # Try the update twice if the first time fails
+    brew update > /dev/null && break
+
+    # If the update fails again after increasing the Git buffer size, exit with error.
```
Comment (TravisEz13) [inline tools/installpsh-osx.sh]: caching the brew cache in travis-ci might help:
```
cache:
  directories:
    - $HOME/Library/Caches/Homebrew
```

**Review by TravisEz13 — APPROVED** [review]

File: tools/installpsh-osx.sh (hunk 2 above)
Comment (daxian-dbw) [reply tools/installpsh-osx.sh]: Thanks Travis! I will add this.

**Comment by adityapatwardhan** [comment]
@daxian-dbw Can you rebase and get #5249

**Comment by daxian-dbw** [comment]
@adityapatwardhan Sorry that I don't understand your comment -- the PR #5249 has been merged for over 2 weeks.

**Comment by daxian-dbw** [comment]
Travis-CI finally passed: https://travis-ci.org/PowerShell/PowerShell/builds/300376520?utm_source=github_status&utm_medium=notification

I will not add code that caches the brew cache yet (Travis suggested that) because I want to verify if the fix can resolve the "Curl error" in `brew update`. We can add that cache optimization later. Issue #5436 opened to track it.

**Comment by adityapatwardhan** [comment]
It seems the fix in #5429 does not fix all the webhook issues. The tests seems to have passed but GitHub doesnot seem it as passed. I will merge.

**Comment by daxian-dbw** [comment]
@adityapatwardhan The webhook fix in that PR is for the build report in our Team channel, not for the Github package. This should be something wrong with the Travis CI.
